File: android/src/main/kotlin/ru/kardenvan/flutter_yandex_ads_plus/ads/YandexAdsSdkFacade.kt
package ru.kardenvan.flutter_yandex_ads_plus.ads

import android.content.Context
import com.yandex.mobile.ads.common.MobileAds
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import ru.kardenvan.flutter_yandex_ads_plus.utils.PluginLogger

object YandexAdsSdkFacade {
    fun initialize(context: Context) {
        MobileAds.initialize(context) {
            PluginLogger.log("Initialized")
        }
    }

    fun setAgeRestrictedUser(value: Boolean) {
        MobileAds.setAgeRestrictedUser(value)

        PluginLogger.log("User age restriction ${if (value) "enabled" else "disabled"}")
    }

    fun setEnableLogging(value: Boolean) {
        MobileAds.enableLogging(value)

        PluginLogger.log("Logging ${if (value) "enabled" else "disabled"}")
    }

    fun setEnableDebugErrorIndicator(value: Boolean) {
        MobileAds.enableDebugErrorIndicator(value)

        PluginLogger.log("Debug error indicator ${if (value) "enabled" else "disabled"}")
    }

    fun setLocationConsent(value: Boolean) {
        MobileAds.setLocationConsent(value)

        PluginLogger.log("Location consent ${if (value) "enabled" else "disabled"}")
    }

    fun setUserConsent(value: Boolean) {
        MobileAds.setUserConsent(value)

        PluginLogger.log("User consent ${if (value) "enabled" else "disabled"}")
    }

    fun getLibraryVersion(): String {
        return MobileAds.getLibraryVersion()
    }
}

File: ios/Classes/ads/YandexAdsSdkFacade.swift
//
//  YandexAdsSDKFacade.swift
//  flutter_yandex_ads_plus
//
//  Created by apple on 24.01.2023.
//

import Foundation
import YandexMobileAds

class YandexAdsSdkFacade {
    static func initialize() {
        YMAMobileAds.initialize()
    }
    
    static func enableLogging(value: Bool) {
        if (value) {
            YMAMobileAds.enableLogging()
            PluginLogger.log(
                message: "Logging enabled."
            )
        } else {
            PluginLogger.log(
                message: "Logging can be disabled only by restarting the app."
            )
        }
    }
    
    static func enableVisibilityErrorIndicator(value: Bool) {
        YMAMobileAds.enableVisibilityErrorIndicator(
            for: value ? YMADeviceType.hardware : []
        )

        PluginLogger.log(
            message: "Debug error indicator \(value ? "enabled" : "disabled")."
        )
    }
    
    static func setLocationTrackingEnabled(value: Bool) {
        YMAMobileAds.setLocationTrackingEnabled(value)
        
        PluginLogger.log(
            message: "Location tracking \(value ? "enabled" : "disabled")."
        )
    }
    
    static func setUserConsent(value: Bool) {
        YMAMobileAds.setUserConsent(value)
        
        PluginLogger.log(
            message: "User consent \(value ? "enabled" : "disabled")"
        )
    }

    static func getSdkVersion() -> String {
        return YMAMobileAds.sdkVersion()
    }
}
